core/storage/base_model/gae_models.py

```diff
@@ -1001,36 +1001,36 @@ def get_multi_versions(cls, entity_id, version_numbers):
                 current version number.
             ValueError. At least one version number is invalid.
         """
-        instances = []
+        if not versions:
+            return []
 
         entity = cls.get(entity_id, strict=False)
         if not entity:
-            raise ValueError('The given entity_id %s is invalid.' % (entity_id))
-        current_version = entity.version
-        max_version = max(version_numbers)
-        if max_version > current_version:
             raise ValueError(
-                'Requested version number %s cannot be higher than the current '
-                'version number %s.' % (max_version, current_version))
+                '%s(id="%s") does not exist' % (cls.__name__, entity_id))
 
-        snapshot_ids = []
-        for version in version_numbers:
-            snapshot_id = cls.get_snapshot_id(entity_id, version)
-            snapshot_ids.append(snapshot_id)
+        max_requested_version = max(versions)
+        if max_requested_version > entity.version:
+            raise ValueError(
+                'Requested version number %s cannot be higher than the current '
+                'version number %s.' % (max_requested_version, entity.version))
 
-        snapshot_models = cls.SNAPSHOT_CONTENT_CLASS.get_multi(snapshot_ids)
-        for snapshot_model in snapshot_models:
+        def reconstitute(snapshot_model, version):
```

Comment:
I would try to avoid nested functions, the previous code seemed clear enough?

Reply:
Done.